tests/test_workers/test_pkg_managers/test_pip.py

```diff
@@ -2971,3 +2999,114 @@ def test_get_index_url_invalid_url():
+    expected_dependency = {"name": name, "version": version, "type": "pip", "dev": dev}
+    dependency = pip._push_downloaded_requirement(req, pip_repo_name, raw_repo_name)
+    mock_upload.assert_called_once_with(pip_repo_name, path)
+    assert dependency == expected_dependency
+
+
+@pytest.mark.parametrize("dev", [True, False])
+@mock.patch("cachito.workers.pkg_managers.pip.upload_raw_package")
+def test_push_downloaded_requirement_vcs(mock_upload, dev):
+    pip_repo_name = "test-pip-hosted"
+    raw_repo_name = "test-pip-raw"
+    name = "eggs"
+    path = "some/path"
+    version = f"vcs:github.com/spam/eggs@{GIT_REF}"
+    raw_component = f"eggs/eggs-external-gitcommit-{GIT_REF}.tar.gz"
+    dest_dir, filename = raw_component.rsplit("/", 1)
+    req = {
+        "package": name,
+        "raw_component_name": raw_component,
+        "path": path,
+        "kind": "vcs",
+        "dev": dev,
+        "host": "github.com",
+        "namespace": "spam",
+        "repo": "eggs",
+        "ref": GIT_REF,
+    }
+    expected_dependency = {"name": name, "version": version, "type": "pip", "dev": dev}
+    dependency = pip._push_downloaded_requirement(req, pip_repo_name, raw_repo_name)
+    mock_upload.assert_called_once_with(raw_repo_name, path, dest_dir, filename, True)
+    assert dependency == expected_dependency
+
+
+@mock.patch("cachito.workers.pkg_managers.pip.get_pip_metadata")
+def test_resolve_pip_no_deps(mock_metadata, tmp_path):
+    mock_metadata.return_value = ("foo", "1.0")
+    request = {"id": 1}
+    pkg_info = pip.resolve_pip(tmp_path, request)
+    expected = {
+        "package": {"name": "foo", "version": "1.0", "type": "pip"},
+        "dependencies": [],
+    }
+    assert pkg_info == expected
+
+
+@mock.patch("cachito.workers.pkg_managers.pip.get_pip_metadata")
+def test_resolve_pip_incompatible(mock_metadata, tmp_path):
+    expected_error = "Could not resolve package metadata: name"
+    mock_metadata.side_effect = CachitoError(expected_error)
+    request = {"id": 1}
+    with pytest.raises(CachitoError, match=expected_error):
+        pip.resolve_pip(tmp_path, request)
+
+
+@mock.patch("cachito.workers.pkg_managers.pip.upload_pypi_package")
+@mock.patch("cachito.workers.pkg_managers.pip.get_pip_metadata")
+@mock.patch("cachito.workers.pkg_managers.pip.download_dependencies")
+def test_resolve_pip(mock_download, mock_metadata, mock_upload, tmp_path):
+    req_file = tmp_path / pip.DEFAULT_REQUIREMENTS_FILE
+    build_req_file = tmp_path / pip.DEFAULT_BUILD_REQUIREMENTS_FILE
+    req_file.write_text("bar==2.1")
+    build_req_file.write_text("baz==0.0.5")
+    mock_metadata.return_value = ("foo", "1.0")
+    mock_download.side_effect = [
+        [{"kind": "pypi", "path": "some/path", "package": "bar", "version": "2.1"}],
+        [{"kind": "pypi", "path": "another/path", "package": "baz", "version": "0.0.5"}],
+    ]
+    request = {"id": 1}
+    pkg_info = pip.resolve_pip(tmp_path, request)
```

Comment:
Could we also have a test where requirement files are passed in as a param?

Reply:
Done